cmd/build.go

```diff
@@ -193,28 +221,30 @@ func GenerateKustomizeResult(config opConfig.Config, kustomizeTemplate template.
 
 	//Write to env files
 	//workflow-config-map.env
-	if yamlFile.HasKeys("artifactRepository.s3.bucket", "artifactRepository.s3.endpoint", "artifactRepository.s3.insecure", "artifactRepository.s3.region") {
-		//Clear previous env file
-		paramsPath := filepath.Join(localManifestsCopyPath, "vars", "workflow-config-map.env")
-		if _, err := files.DeleteIfExists(paramsPath); err != nil {
-			return "", err
-		}
-		paramsFile, err := os.Create(paramsPath)
-		if err != nil {
-			return "", err
-		}
-		var stringToWrite = fmt.Sprintf("%v=%v\n%v=%v\n%v=%v\n%v=%v\n",
-			"artifactRepositoryBucket", flatMap["artifactRepositoryS3Bucket"],
-			"artifactRepositoryEndpoint", flatMap["artifactRepositoryS3Endpoint"],
-			"artifactRepositoryInsecure", flatMap["artifactRepositoryIS3nsecure"],
-			"artifactRepositoryRegion", flatMap["artifactRepositoryS3Region"],
-		)
-		_, err = paramsFile.WriteString(stringToWrite)
-		if err != nil {
-			return "", err
+	if yamlFile.HasKey("artifactRepository.s3") {
```

Comment:
This is a bit confusing, I see a check here for s3, but nothing for `gcs`. Maybe explain why we only check for `s3` here?

Reply:
Also, should we just check `artifactRepositoryConfig.S3 != nil` here too?

Reply:
Agreed, updated.